selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/CreateStackTest.java

```diff
@@ -0,0 +1,141 @@
+      "services:\n"
+          + " db:\n"
+          + "  image: eclipse/mysql\n"
+          + "environment:\n"
+          + "  MYSQL_ROOT_PASSWORD: password\n"
+          + "MYSQL_DATABASE: petclinic\n"
+          + "MYSQL_USER: petclinic\n"
+          + "MYSQL_PASSWORD: password\n"
+          + "\b\bmem_limit: 1073741824\n"
+          + "\b\bdev-machine:\n"
+          + "  image: eclipse/ubuntu_jdk8\n"
+          + "mem_limit: 2147483648\n"
+          + "depends_on:\n"
+          + "  - db";
+  private static final String DOCKERIMAGE_RECIPE = "eclipse/ubuntu_jdk8";
+  private static final String DOCKERFILE_RECIPE = "FROM eclipse/ubuntu_jdk8";
+  private static final String KUBERNETES_RECIPE =
+      "---\n"
+          + "kind: List\n"
+          + "items:\n"
+          + "-\n"
+          + "  apiVersion: v1\n"
+          + "kind: Pod\n"
+          + "metadata:\n"
+          + "  name: ws\n"
+          + "\b\bspec:\n"
+          + "  containers:\n"
+          + "  -\n"
+          + "  image: eclipse/ubuntu_jdk8\n"
+          + "name: dev-machine";
+
+  @Inject private Stacks stacks;
+  @Inject private Dashboard dashboard;
+  @Inject private StackDetails stackDetails;
+  @Inject private BuildStackFromRecipe buildStackFromRecipe;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    dashboard.open();
+  }
+
+  @BeforeMethod
+  public void openStacksListPage() {
+    dashboard.waitDashboardToolbarTitle();
+    dashboard.selectStacksItemOnDashboard();
+    stacks.waitToolbarTitleName();
+
+    stacks.clickOnAddStackButton();
+    buildStackFromRecipe.waitCreateStackDialogVisible();
+  }
+
+  @AfterClass
+  public void deleteCreatedStacks() {
+    dashboard.selectStacksItemOnDashboard();
+    stacks.waitToolbarTitleName();
+
+    stacks.selectAllStacksByBulk();
+    stacks.clickOnDeleteStackButton();
+    stacks.clickOnDeleteDialogButton();
+
+    dashboard.waitNotificationMessage("Selected stacks have been successfully removed.");
+    dashboard.waitNotificationIsClosed();
+  }
+
+  @Test(groups = {TestGroup.DOCKER})
+  public void createStackFromComposeRecipe() {
+    createStack("compose", COMPOSE_RECIPE);
+  }
+
+  @Test
+  public void createStackFromDockerimageRecipe() {
+    createStack("dockerimage", DOCKERIMAGE_RECIPE);
+  }
+
+  @Test(groups = {TestGroup.DOCKER})
+  public void createStackFromDockerfileRecipe() {
+    createStack("dockerfile", DOCKERFILE_RECIPE);
+  }
+
+  @Test(groups = {TestGroup.OPENSHIFT})
+  public void createStackFromKubernetesRecipe() {
+    createStack("kubernetes", KUBERNETES_RECIPE);
+  }
+
+  @Test(groups = {TestGroup.OPENSHIFT})
+  public void createStackFromOpenshiftRecipe() {
+    createStack("openshift", KUBERNETES_RECIPE);
+  }
+
+  private void createStack(String stackName, String recipe) {
+    String name = generate(stackName, 5);
+
+    buildStackFromRecipe.selectTabByName(stackName);
+
+    // wait for editor to ready for typing after tab selection
+    WaitUtils.sleepQuietly(1);
```

Comment:
It would better to get rid of sleep.

Reply:
We need to wait to ensure that the selected tab is active.